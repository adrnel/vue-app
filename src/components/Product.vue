<template>
  <div>
    <img :src="product.image">
    <h1>Name: {{product.name}}</h1>
    <h3>Price: {{product.currency}}{{product.price}}</h3>
    <p>Description: {{product.long_desc}}</p>
    <button v-on:click="addToBasket">Add to Basket</button>
  </div>
</template>

<script>
import { mapState } from 'vuex'
export default {
  name: 'product',
  data () {
    return {
    }
  },
  props:{
    product: {
      type: Object
    }
  },
  methods: {
    addToBasket (product) {
      this.$store.commit('addToBasket', this.product)
    }
  },
  computed: {
    ...mapState([
      'basket'
    ])
  },
};

</script>

<style lang="scss" scoped>
  img {
    width: 100px;
    height: 100px;
  }
  .container{
    display: flex;
  }
  .item{
    flex-grow: 1
  }
  .column {
    margin: 20px;
  }
</style>