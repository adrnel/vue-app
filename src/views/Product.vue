<template>
  <div class="product">
    <ProductList></ProductList>
  </div>
</template>

<script>
import ProductList from '../components/ProductList.vue'
export default {
  name: 'product',
  data () {
    return {
    }
  },
  components: {
    ProductList
  },
   methods: {
   }
};
</script>

<style lang="scss">
</style>