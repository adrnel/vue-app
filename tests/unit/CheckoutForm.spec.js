import { shallow } from '@vue/test-utils';
import Product from '../../src/components/Product.vue';

describe('Product.vue', () => {
  it('renders props.msg when passed', () => {
    const wrapper = shallow(Product, {propsData: { product: {
      name: 'Cube',
      currency: '£',
      price: 4.99,
      image: 'cube.jpg',
      long_desc: 'Cube Text'
    }}})
    const componentText = "Name: Cube Price: £4.99 Description: Cube Text Add to Basket"
    expect(wrapper.text()).toMatch(componentText);
  });
  it('matches snapshot', () => {
    const wrapper = shallow(Product, {propsData: { product: {
      name: 'Cube',
      currency: '£',
      price: 4.99,
      image: 'cube.jpg',
      long_desc: 'Cube Text'
    }}})
    expect(wrapper.html()).toMatchSnapshot()
  })

});
