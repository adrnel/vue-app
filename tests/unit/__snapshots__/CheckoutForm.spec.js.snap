// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product.vue matches snapshot 1`] = `
<div>
    <img src="cube.jpg">
    <h1>Name: Cube</h1>
    <h3>Price: Â£4.99</h3>
    <p>Description: Cube Text</p>
    <button>Add to Basket</button>
</div>
`;
